tools/InterfaceGenerator/generator/generators/SmartFactoryBase.py

```diff
@@ -840,22 +840,36 @@ def _gen_schema_item_decl_code(self, param, member_name, default_value):
                     [[u"bool", None if param.default_value is None
                       else u"true" if param.default_value is True else u"false"]]))
         elif type(param) is Integer:
-            if not param.max_value or param.max_value and param.max_value < 2 ** 31:
+            if (param.min_value is not None and param.min_value >= 0) and (param.max_value is None or param.max_value >= 2 ** 32):
+                code = self._impl_code_integer_item_template.substitute(
+                    type=u"uint64_t",
+                    params=self._gen_schema_item_param_values(
+                        [[u"uint64_t", param.min_value],
+                         [u"uint64_t", param.max_value],
+                         [u"uint64_t", param.default_value]]))
+            elif (param.min_value is not None and param.min_value >= 0) and (param.max_value is not None and param.max_value < 2 ** 32):
+                code = self._impl_code_integer_item_template.substitute(
+                    type=u"uint32_t",
+                    params=self._gen_schema_item_param_values(
+                        [[u"uint32_t", param.min_value],
+                         [u"uint32_t", param.max_value],
+                         [u"uint32_t", param.default_value]]))
```

Comment:
```suggestion
            if param.min_value is not None and param.min_value >= 0:
                if param.max_value is None or param.max_value >= 2 ** 32:
                    code = self._impl_code_integer_item_template.substitute(
                        type=u"uint64_t",
                        params=self._gen_schema_item_param_values(
                            [[u"uint64_t", param.min_value],
                             [u"uint64_t", param.max_value],
                             [u"uint64_t", param.default_value]]))
                else:
                    code = self._impl_code_integer_item_template.substitute(
                        type=u"uint32_t",
                        params=self._gen_schema_item_param_values(
                            [[u"uint32_t", param.min_value],
                             [u"uint32_t", param.max_value],
                             [u"uint32_t", param.default_value]]))
```
To simplify the expression a bit

Reply:
@jacobkeeler fixed in https://github.com/smartdevicelink/sdl_core/pull/3750/commits/4e8df45491863d74a54320049c9cf671bf1baaf4